src/tools/utils.js

```diff
@@ -62,7 +62,71 @@ export function dumpJSON(obj, spacing = 0) {
   return JSON.stringify(obj, null, spacing);
 }
 
-export function calcChanges(assets, existing, identifiers = [ 'id', 'name' ]) {
+/**
+ * @template T
+ * @param {typeof import('./auth0/handlers/default').default} assetType
+ * @param {T} desiredAssetState
+ * @param {T} currentAssetState
+ * @param {string[]} [objectFields=[]]
+ * @param {boolean} [allowDelete=false]
+ * @returns T
+ */
+function processChangedObjectFields(assetType, desiredAssetState, currentAssetState, objectFields = [], allowDelete = false) {
```

Comment:
Can we rename assetType to handler, for me assetType is inferring a string.

Reply:
Fixed in cb3cca4.